tests/unit/s2n_server_new_session_ticket_test.c

```diff
@@ -693,32 +722,65 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_config_free(config));
         }
 
-        /* Can't write ticket larger than allowed size of a PSK identity */
+        /* Test that the client rejects tickets with invalid ticket_lifetime */
         {
+            const size_t lifetime_size = sizeof(uint32_t);
+            const uint8_t *nst_data_without_lifetime = nst_data + lifetime_size;
+            const size_t nst_data_without_lifetime_size = sizeof(nst_data) - lifetime_size;
+
             struct s2n_config *config = s2n_config_new();
             EXPECT_NOT_NULL(config);
             EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(config, 1));
             EXPECT_SUCCESS(s2n_config_set_session_ticket_cb(config, s2n_test_session_ticket_cb, NULL));
-            EXPECT_SUCCESS(s2n_setup_test_ticket_key(config));
 
-            struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER);
-            EXPECT_NOT_NULL(server_conn);
-            EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
-            server_conn->actual_protocol_version = S2N_TLS13;
-            server_conn->secure.cipher_suite = &s2n_tls13_aes_256_gcm_sha384;
-            EXPECT_SUCCESS(s2n_connection_set_server_max_early_data_size(server_conn, 10));
+            /**
+             *= https://tools.ietf.org/rfc/rfc8446#section-4.6.1
+             *= type=test
+             *# The value of zero indicates that the
+             *# ticket should be discarded immediately.
+             */
+            {
+                struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+                EXPECT_NOT_NULL(conn);
+                EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+                conn->actual_protocol_version = S2N_TLS13;
 
-            /* Set context to be UINT16_MAX */
-            uint8_t early_data_context[UINT16_MAX] = { 0 };
-            EXPECT_SUCCESS(s2n_connection_set_server_early_data_context(server_conn,
-                    early_data_context, sizeof(early_data_context)));
+                DEFER_CLEANUP(struct s2n_stuffer input = { 0 }, s2n_stuffer_free);
+                EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&input, sizeof(nst_data)));
+                EXPECT_SUCCESS(s2n_stuffer_write_uint32(&input, 0));
+                EXPECT_SUCCESS(s2n_stuffer_write_bytes(&input, nst_data_without_lifetime, nst_data_without_lifetime_size));
 
-            DEFER_CLEANUP(struct s2n_stuffer stuffer = { 0 }, s2n_stuffer_free);
-            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&stuffer, 0));
+                EXPECT_OK(s2n_tls13_server_nst_recv(conn, &input));
+                EXPECT_EQUAL(s2n_connection_get_session_length(conn), 0);
```

Comment:
Assuming that this is the line which indicates that the ticket was currently rejected, please add a comment to this line explaining that for future reviewers.